lib/dry/schema/step.rb

```diff
@@ -17,53 +20,34 @@ class Step
       attr_reader :executor
 
       # @api private
-      class Scoped
-        # @api private
-        attr_reader :path
-
-        # @api private
-        attr_reader :step
-
-        # @api private
-        def initialize(path, step)
-          @path = Path[path]
-          @step = step
-        end
-
-        # @api private
-        def scoped(new_path)
-          self.class.new(Path[[*new_path, *path]], step)
-        end
-
-        # @api private
-        def call(result)
-          result.at(path) do |scoped_result|
-            output = step.(scoped_result).to_h
-            target = Array(path)[0..-2].reduce(result) { |a, e| a[e] }
-
-            target.update(path.last => output)
-          end
-        end
-      end
+      attr_reader :path
 
       # @api private
-      def initialize(type:, name:, executor:)
+      def initialize(type:, name:, executor:, path: EMPTY_PATH)
```

Comment:
No `#dup` here since we want to use blazing fast `#equal?` below instead of slow `#eql?` or whatever else.
We're not mutating path anyway and it's frozen, so :thisisfine:.

Knowing about implementation details we should froze `EMPTY_PATH.keys`, but let's pretend that we don't know it, if someone wanna mess up with internals of our private constant — all bets are off.